### ISSUE TITLE
Poor Validation Performance After Retraining, Despite Decreasing Loss

### ISSUE THREAD

**Issue by MLS2021** [cvlab-stonybrook/ZoomLDM#6]
After preparing the dataset as suggested, I retrained the model using your configuration. Since I’m using an RTX 4090 GPU, I set the batch_size to 32.
Initially, I trained on a small dataset (~3000 images), and the training loss quickly dropped to around 0.05. However, the validation performance was very poor (see the attached image)：

<img width="325" height="601" alt="Image" src="https://github.com/user-attachments/assets/ee374f82-1225-4ac9-8dcf-0500799bbb05" />

To stay aligned with your original setup, I added gradient accumulation:
`accumulate_grad_batches: 13`
In later runs, I encountered NaN losses, so I applied the following settings to stabilize training:
```
precision: bf16
accumulate_grad_batches: 13
gradient_clip_val: 1.0
```

In the most recent training session (2 epochs), the loss stayed around 0.15 without further decrease. It took around 10 hours, and I stopped training to run validation — the results were still disappointing.
train log：
```
Epoch 0, global step 1792: val/loss was not in top 3
Epoch 1: 100%|█| 23301/23301 [5:42:00<00:00,  1.14it/s, loss=0.144, v_num=0, train/loss_simple_step=0.153, train/loss_vlb_step=0.00122, train/loss_step=0.153, global_step=3585.0, lr_abs=0.000416, train/loss_simple_epochAverage Epoch time: 20520.49 seconds
Average Peak memory 16810.99MiB
Epoch 1, global step 3585: val/loss was not in top 3
Epoch 2: 100%|█| 23301/23301 [5:42:42<00:00,  1.13it/s, loss=0.162, v_num=0, train/loss_simple_step=0.139, train/loss_vlb_step=0.00395, train/loss_step=0.139, global_step=5378.0, lr_abs=0.000416, train/loss_simple_epochAverage Epoch time: 20562.66 seconds
Average Peak memory 16817.29MiB
Epoch 2, global step 5378: val/loss was not in top 3
Epoch 3:   2%| | 401/23301 [05:55<5:37:58,  1.13it/s, loss=0.158, v_num=0, train/loss_simple_step=0.200, train/loss_vlb_step=0.00182, train/loss_step=0.200, global_step=5409.0, lr_abs=0.000416, train/loss_simple_epoch=0^CProcess ForkProcess-27:
```
val result：
<img width="844" height="406" alt="Image" src="https://github.com/user-attachments/assets/3b34ebc8-6c74-44a0-ad7b-003475b2aebf" />

I’m confident the dataset is not the issue — using your pretrained weights on the same data yielded good results. So I suspect the problem lies in the training process or hyperparameter setup.

<img width="826" height="418" alt="Image" src="https://github.com/user-attachments/assets/d91ae0db-71f6-4716-b801-d5f4217cd851" />

I’ve tried several approaches but haven’t been able to improve the validation performance. I would really appreciate any advice or debugging suggestions. My current config is almost identical to yours, except for the batch_size setting.

**Comment by srikarym** [comment]
I believe the learning rate (1e-6) is too low, can you try a higher value?

**Comment by MLS2021** [comment]
yeah,I will try again. But as shown in the picture, I think model didn’t learned the condition of ssl_feat. Regarding the satellite images, how many steps did you train at that time?

**Comment by kostino** [comment]
I think it's probably an issue with the number of train steps. Your dataset has ~3k images and you trained for 2 epochs with a batch size of 32, this means ~200 steps in total (even less with grad accumulation). With the config that you use the first 1k steps are warmup steps. I think it just needs more training; we trained our NAIP model for more than 50k steps.
